tokio/src/runtime/thread_pool/worker.rs

```diff
@@ -442,6 +454,18 @@ impl Context {
     }
 
     fn park(&self, mut core: Box<Core>) -> Box<Core> {
+        if let Some(f) = &self.worker.shared.before_park {
+            f();
+            // If `f` scheduled any tasks on the local queue, don't park - the extra
+            // work should be done now.
+            if core.lifo_slot.is_some() || !core.run_queue.has_tasks() {
+                if let Some(f) = &self.worker.shared.after_unpark {
+                    f();
+                }
+                return core;
+            }
+        }
```

Comment:
This seems fragile. Future changes could easily break this without noticing.

Reply:
I can refactor the check into a separate function - I've done it like this because today, `spawn` will always fill the `lifo_slot`, but I can see a world in which it doesn't do that, and just adds tasks to the `run_queue`.

Reply:
Moved the check to the transition function, as is done for unpark checks.